### PR TITLE
EVG-14258: Finalize Child patch intents right away if parentStatus = "" 

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: units/patch_intent.go [hunk 1]
```diff
@@ -493,7 +493,10 @@ func (j *patchIntentProcessor) processTriggerAliases(ctx context.Context, p *pat
 	}
 
 	for _, intent := range triggerIntents {
-		if err := j.env.RemoteQueue().Put(ctx, NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)); err != nil {
+		job := NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)
+		job.Run(ctx)
+
+		if err := job.Error(); err != nil {
 			return errors.Wrap(err, "problem enqueueing child patch processing")
```
Comment (ybrill) [inline units/patch_intent.go:500]: the error is now not about _enqueueing_ the processing, but rather in the actual processing

File: units/patch_intent.go [hunk 2]
```diff
@@ -493,7 +493,10 @@ func (j *patchIntentProcessor) processTriggerAliases(ctx context.Context, p *pat
 	}
 
 	for _, intent := range triggerIntents {
-		if err := j.env.RemoteQueue().Put(ctx, NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)); err != nil {
+		job := NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)
```
Comment (ybrill) [inline units/patch_intent.go:496]: I was thinking we could do this _only_ in the case of `parentStatus == ""`
That way we could have the performance advantages of separating the processing at least in the case where the child patch is anyways waiting for the parent patch to complete.

Comment (malikchaya2) [reply units/patch_intent.go:496]: ah yes, I did that, but looks like I didn't save my file before committing. I'll fix. Thanks for pointing it out!

File: units/patch_intent.go (hunk 1 above)
Comment (malikchaya2) [reply units/patch_intent.go:500]: good point, thanks!

File: units/patch_intent.go [hunk 3]
```diff
@@ -493,11 +493,23 @@ func (j *patchIntentProcessor) processTriggerAliases(ctx context.Context, p *pat
...
-		if err := j.env.RemoteQueue().Put(ctx, NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)); err != nil {
-			return errors.Wrap(err, "problem enqueueing child patch processing")
+		triggerIntent, ok := intent.(*patch.TriggerIntent)
+		if !ok {
+			return errors.Errorf("intent '%s' didn't not have expected type '%T'", intent.ID(), intent)
 		}
-	}
 
+		if triggerIntent.ParentStatus == "" {
+			job := NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)
```
Comment (ybrill) [inline units/patch_intent.go:502]: The creation of the job could be factored out of the if/else

File: units/patch_intent.go [hunk 4]
```diff
@@ -493,11 +493,23 @@ func (j *patchIntentProcessor) processTriggerAliases(ctx context.Context, p *pat
...
 	for _, intent := range triggerIntents {
-		if err := j.env.RemoteQueue().Put(ctx, NewPatchIntentProcessor(mgobson.ObjectIdHex(intent.ID()), intent)); err != nil {
-			return errors.Wrap(err, "problem enqueueing child patch processing")
+		triggerIntent, ok := intent.(*patch.TriggerIntent)
+		if !ok {
+			return errors.Errorf("intent '%s' didn't not have expected type '%T'", intent.ID(), intent)
 		}
-	}
 
+		if triggerIntent.ParentStatus == "" {
```
Comment (ybrill) [inline units/patch_intent.go:501]: maybe we should leave a comment why we want to process it right away in this case

**Review by ybrill — APPROVED** [review]

**Comment by malikchaya2** [comment]
evergreen merge

**Pull request merged**